### ISSUE TITLE
`node-postgres` (pg) errors with misleading `P2010 PrismaClientKnownRequestError` when using `@prisma/adapter-pg` with SSL (`?sslmode=require`)

### ISSUE THREAD

**Issue by exAspArk** [prisma/prisma#23390]
### Bug description

I'm using:

* Supabase with Enforced SSL
* `pg` as the driver adapter
* `?sslmode=require` in the DB connection URL

And seeing this error:

```
PrismaClientKnownRequestError: 
Invalid `prisma.todo.findMany()` invocation:

Raw query failed. Code: `InvalidArg`. Message: `invalid type: unit value, expected a string`
 code: "P2010"
```

### How to reproduce

Follow these instructions https://www.prisma.io/docs/orm/overview/databases/postgresql#using-the-node-postgres-driver and use the `sslmode=require` connection argument

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["driverAdapters"]
}

datasource db {
  provider = "postgresql"
  url      =  env("DATABASE_URL")
}

model Todo {
  @@map("todos")

  id        Int      @id @default(autoincrement())
  task      String   @db.VarChar(255)
  isCompleted Boolean  @default(false)
}
```

```ts
import { Pool } from 'pg'
import { PrismaPg } from '@prisma/adapter-pg'
import { PrismaClient } from '@prisma/client'

const connectionString = `${process.env.DATABASE_URL}`

const pool = new Pool({ connectionString })
const adapter = new PrismaPg(pool)
const prisma = new PrismaClient({ adapter })
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL 15.1
- Node.js version: v20.8.0

### Prisma Version

```
5.9.1
```

**Comment by Jolg42** [comment]
I can reproduce this with the information provided here on a recent dev version (`5.11.0-dev.48`)

With `DEBUG=*` logs, I found out that the underlying error message is:
```
prisma:driver-adapter:pg Error in performIO: '{\n  "code": "SELF_SIGNED_CERT_IN_CHAIN"\n}'
```

**Comment by Jolg42** [comment]
@exAspArk We need to improve the error handling here, thanks for this issue 🙌🏼 

For the SSL error, this is not a problem on Prisma's side.
Because in your case the certificates are self-signed, I can recommend you to switch to `no-verify`, which means the connection would be encrypted, but the certificate will not be verified (see more info https://stackoverflow.com/a/31862256/1345244)

If you want to use `sslmode=verify-full` and equivalents you would need to pass the `sslrootcert` parameter.
Supabase has a "Download certificate" button here https://supabase.com/docs/guides/database/connecting-to-postgres#connecting-with-ssl

You can then use a connection string like this: 
(here I used a relative path for the certificate, it can also be an absolute path)
```
  'postgres://postgres.id:password@aws-0-eu-central-1.pooler.supabase.com:5432/postgres?sslmode=verify-full&sslrootcert=./prod-ca-2021.crt'
```

**Comment by exAspArk** [comment]
Wow, you're so fast! Thanks a lot, @Jolg42! 💛 

I'm going to close the issue. Hopefully, people will be able to find it by the current error message in the meantime.

**Comment by Jolg42** [comment]
Thanks @exAspArk I prefer to keep it open, as I renamed the issue. I think we can use it to track this as a bug related to the error handling.

**Comment by jremi** [comment]
Hi guys, just wanted to say thank you for this thread. It saved me alot of time. I spent about an hour testing my local environment against a newly deployed Digital Ocean setup (App Platform/Managed Postgresql database). Kept failing with the same exact issue.

For now i ended up just using the `sslmode=no-verify` option to resolve this.

Many of the cloud providers (e.g: Digital Ocean Databases) provide pre-filled connection strings that you can just copy / paste. In this example, they tack on the `sslmode=require` ... So i think many people naturally just copy/paste this and assume it will be fine without any additional configuration.

Anyway, thanks again.

**Comment by SevInf** [comment]
Error handing for this case have been fixed and Prisma 5.12.0 and above would now correctly preserve original message.
